### ISSUE TITLE
Performance issues compared to other minisat bindings

### ISSUE THREAD

**Issue by Gbury** [ujamjar/sattools#6]
In my quest to benchmarks sat solvers in ocaml, I got all ocaml libs on opam that expose sat solver interfaces, and made a small ocaml program to compare them. The result is at https://github.com/Gbury/sat-bench

One surprising result was the comparison between the `sattools` minisat bindings and the two other bindings: `ocaml-minisat` ( https://github.com/c-cube/ocaml-minisat ), and `ocaml-sat-solvers` ( https://github.com/tcsprojects/ocaml-sat-solvers ). You can see the results at https://github.com/Gbury/sat-bench/blob/master/problems/pigeon/results

The problem is that the minisat bindings in `sattools` are **a lot slower** than the two other bindings, though still faster than the pure ocaml sat solver `msat`, as expected. It might be interesting to investigate the source of the differences, particularly considering the performance differences on the last problems.

This might be of interest to @c-cube, author of `ocaml-minisat`, and @tcsprojects, author of `ocaml-sat-solvers`.

**Comment by yallop** [comment]
It seems that `ocaml-sat-solvers` wraps `ocaml-minisat`.  That'd explain why the numbers for those two are so similar.

The difference between `ocaml-minisat` and `sattools` may be due to different versions of the `minisat` library; `ocaml-minisat` includes the source for the C library Minisat-C-1.14.1, while `sattools` binds to the system-installed C++ library.

**Comment by andrewray** [comment]
Yes, the minisat libraries used are completely different - sattools-minisat uses the C++ minisat2 library.

The performance issue here seems to be related to the use of the *luby* option (on by default with minisat2).  You can run the test using the minisat command line and the following fixes the problem.

```
$ minisat -no-luby <cnf-file>
``` 

It wouldn't surprise me if, on other benchmarks, the situation is reversed.
